wip-0009.md

```diff
@@ -0,0 +1,83 @@
+<pre>
+    WIP: WIP-0009
+    Layer: Consensus (hard fork)
+    Title: Adjust mining probability
+    Authors: Mario Cao <mario@witnet.foundation>
+             Gorka Irazoqui <gorka.irazoki@gmail.com>
+    Discussions-To: `#dev-general` channel on Witnet Community's Discord server
+    Status: Draft
+    Type: Standards Track
+    Created: 2021-01-26
+    License: BSD-2-Clause
+</pre>
+
+
+## Abstract
+
+This proposal adjusts the mining algorithm to limit the probability of non-reputed and non-active nodes mining a block.
+
+The goal of this proposal is to mitigate potential network abuses that could be carried by malicious users controlling a large number of nodes. This could pose a serious threat to the network availability and security.
+
+
+## Motivation and Rationale
+
+after more than 3 months of the Witnet mainnet launch, the network has faced episodes of a bigger transaction latency than expected. In periods with a mempool full of transactions, blocks not using the complete space appeared in the Witnet network. This reduces the efficiency of the system, thus increasing indirectly the transaction latency perceived by users. Under a more critical scenario, this problem could pose an availability threat in terms of a [denial-of-service attack][DoS], in which a large number of malicious nodes could be mining empty blocks or, even, endanger the network stability by affecting internal network metrics (e.g. Active Reputation Set).
+
+A thorough investigation concluded that the majority of these blocks (if not all) were produced by non-reputed and non-active nodes (i.e. newcomers to the network). Regardless of the intentionality of this behaviour, these blocks were appearing in a much higher volume than the newcomer mining probability that was designed in the protocol.
+
+The current mining algorithm can be summarized as follows:
+
+1. Miners build a candidate block.
+2. Miners compute their eligibility by using the [VRF]-based [RandPoE] algorithm with by using the ARS census as the difficulty parameter (labelled as N).
+3. If eligible (and no better candidate was received), miners will broadcast their candidate blocks.
+
+Upon reception of multiple candidates, the winner will be selected as follows:
+1. Candidate blocks from reputed nodes are prioritized over those from non-reputed nodes.
+2. If there are still multiple candidates, the one with the lowest VRF is chosen.
```

Comment:
For the sake of completeness, is it worth mentioning that in case of a tie, block hashes are preferred in lexicographical order?

Reply:
The VRF output is a number... 
I suggest: "2. If there are still multiple candidates, the one with the lowest VRF value is chosen."

Reply:
Yeah, but I mean that this description doesn't really match what I see in the code (it's missing "slots" and block hash comparison).

Reply:
There were left out on purpuse for the sake of simplicity. Nevertheless, we added 2 more steps. :)